lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -18,22 +18,24 @@
 MAXIMUM_DATE = utc.localize(datetime.max)
 
 
-class HiddenContentTransformer(FilteringTransformerMixin, BlockStructureTransformer):
+class HiddenContentTransformer(BlockStructureTransformer):
     """
     A transformer that enforces the hide_after_due field on
     blocks by removing children blocks from the block structure for
-    which the user does not have access. The due and hide_after_due
-    fields on a block is percolated down to its descendants, so that
+    which the user does not have access. The hide_after_due
+    field on a block is percolated down to its descendants, so that
     all blocks enforce the hidden content settings from their ancestors.
 
     For a block with multiple parents, access is denied only if
     access is denied from all its parents.
 
     Staff users are exempted from hidden content rules.
+
+    IMPORTANT: Must be run _after_ the DateOverrideTransformer from edx-when
+    in case the 'due' date on a block has been shifted for a user.
     """
-    WRITE_VERSION = 2
-    READ_VERSION = 2
-    MERGED_DUE_DATE = 'merged_due_date'
+    WRITE_VERSION = 3
+    READ_VERSION = 3
```

Comment:
I think this is required, but if someone wants to sanity check me... https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/block_structure/transformer.py#L16